code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -1462,3 +1463,123 @@
 		to_chat(src, "<span class='xenowarning'><b>You can now savage your victims again.</b></span>")
 		playsound(src, "xeno_newlarva", 100, 0, 1)
 		update_action_button_icons()
+
+
+/mob/living/carbon/Xenomorph/proc/spray_turfs(list/turflist)
+	set waitfor = 0
+
+	if(isnull(turflist))
+		return
+	var/turf/prev_turf
+	var/distance = 0
+
+	turf_loop:
+		for(var/turf/T in turflist)
+			distance++
+
+			if(!prev_turf && turflist.len > 1)
+				prev_turf = get_turf(src)
+				continue //So we don't burn the tile we be standin on
+
+			if(T.density || istype(T, /turf/open/space))
+				break
+			if(distance > 7)
+				break
+
+			if(locate(/obj/structure/girder, T))
+				break //Nope.avi
+
+			var/obj/machinery/M = locate() in T
+			if(M)
+				if(M.density)
+					break
+
+			if(prev_turf && LinkBlocked(prev_turf, T))
+				break
+
+			for(var/obj/structure/barricade/B in T)
+				B.health -= rand(45, 60) + 8 * upgrade
+				B.update_health(TRUE)
+				if(prev_turf)
+					if(get_dir(B, prev_turf) & B.dir)
+						break turf_loop
+
+			prev_turf = T
+			splat_turf(T)
+			sleep(2)
+
+
+/mob/living/carbon/Xenomorph/proc/splat_turf(var/turf/target)
+	if(!istype(target) || istype(target,/turf/open/space))
+		return
+
+	for(var/obj/effect/xenomorph/spray/S in target) //No stacking spray!
+		cdel(S)
+	new /obj/effect/xenomorph/spray(target)
+	for(var/mob/living/carbon/M in target)
+		if(ishuman(M) || ismonkey(M))
+			if((M.status_flags & XENO_HOST) && istype(M.buckled, /obj/structure/bed/nest))
+				continue //nested infected hosts are not hurt by acid spray
+			M.adjustFireLoss(rand(25, 38) + 4 * upgrade)
+			to_chat(M, "<span class='xenodanger'>\The [src] showers you in corrosive acid!</span>")
+			if(!isYautja(M))
+				M.emote("scream")
+				M.KnockDown(1)
+
+/mob/living/carbon/Xenomorph/proc/acid_spray(atom/T, plasmacost = 150, acid_delay = src.acid_delay)
```

Comment:
`src` is implied, so here it's basically saying `acid_delay = acid_delay`
Either change the name of the var or remove it in favor of the already existing var defined for `Xenomorph`.